src/content/1.7/development/architecture/domain/cqrs.md

```diff
@@ -42,10 +42,10 @@ Fortunately, by implementing CQRS it allows PrestaShop to quickly build new API,
 2. A `Command` describes a __single__ action. It DOES NOT perform it.
 3. A `Command` receives only primitive types on input (int, float, string, bool and array).
 4. For every `Command` there MUST be at least one `CommandHandler` whose role is to execute that `Command`.
-5. A `CommandHandler` MUST be placed in the `Adapter` namespace as long as it has legacy dependencies.
-6. A `CommandHandler` SHOULD NOT return anything on success, and SHOULD throw a typed `Exception` on failure. The "no return on success" rule can be broken only when creating entities. 
-7. A `CommandHandler` MUST use the existing `ObjectModel` for writes as long as those classes exist.
-8. A `CommandHandler` MUST implement an interface containing a single public method like this: 
+5. A `CommandHandler` is just a port to a core, therefore it MUST NOT handle the domain logic itself, but orchestrate the required [services](services.md) instead.
```

Comment:
```suggestion
5. A `CommandHandler` is just a port to a core, therefore it MUST NOT handle the domain logic itself, but orchestrate the required [services](services.md) instead.
```

Unless the domain logic is 4 lines long and is really unique? 😅

Reply:
yeah, but I don't think its worth adding it in docs, we will decide during PR, as such cases should be very rare :smile: 
I could only change MUST NOT -> SHOULD NOT maybe?

Reply:
No you're right

Reply:
SHOULD NOT is appropriate.
The CommandHandler is not "just a port", I'd rather define it as an intermediate controller.